core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -896,12 +896,33 @@ static unsigned int get_physical_addr_size_bits(void)
 	if (CFG_CORE_ARM64_PA_BITS <= 44)
 		return TCR_PS_BITS_16TB;
 
-	/* Physical address can't exceed 48 bits */
-	COMPILE_TIME_ASSERT(CFG_CORE_ARM64_PA_BITS <= 48);
 	/* CFG_CORE_ARM64_PA_BITS <= 48 */
 	return TCR_PS_BITS_256TB;
 }
 
+static unsigned int get_physical_addr_size_bits(void)
+{
+	const unsigned int size_bits = read_id_aa64mmfr0_el1() &
+				       ID_AA64MMFR0_EL1_PARANGE_MASK;
+	unsigned int b = 0;
+
+	if (IS_ENABLED(CFG_AUTO_MAX_PA_BITS))
+		return size_bits;
+
+	b = get_hard_coded_pa_size_bits();
+	assert(b <= size_bits);
+	return b;
+}
+
+unsigned int core_mmu_arm64_get_pa_width(void)
+{
+	const uint8_t map[] = { 32, 36, 40, 42, 44, 48, 52, };
+	unsigned int size_bits = get_physical_addr_size_bits();
+
+	size_bits = MIN(size_bits, ARRAY_SIZE(map) - 1);
```

Comment:
`assert(size_bits < ARRAY_SIZE(map))`?

Reply:
I'm trying to be future-proof. If Arm adds another value it's nice not to cause any breakage.